clients/unshaded/src/main/java/tachyon/client/file/URIStatus.java

```diff
@@ -0,0 +1,196 @@
+
+  /**
+   * @return a list of block ids belonging to the file, empty for directories, immutable
+   */
+  public List<Long> getBlockIds() {
+    return mInfo.getBlockIds();
+  }
+
+  /**
+   * @return the default block size for this file, 0 for directories, immutable
+   */
+  public long getBlockSizeBytes() {
+    return mInfo.getBlockSizeBytes();
+  }
+
+  /**
+   * @return the epoch time the entity referenced by this uri was created, immutable
+   */
+  public long getCreationTimeMs() {
+    return mInfo.getCreationTimeMs();
+  }
+
+  /**
+   * @return the unique identifier of the entity referenced by this uri used by Tachyon servers,
+   *         immutable
+   */
+  public long getFileId() {
+    return mInfo.getFileId();
+  }
+
+  /**
+   * @return the group that owns the entity referenced by this uri, mutable
+   */
+  public String getGroupName() {
+    return mInfo.getGroupName();
+  }
+
+  /**
+   * @deprecated This method is temporary
+   *
+   * @return the backing {@link FileInfo} object
+   */
+  // TODO(calvin): Remove this method
+  @Deprecated
+  public FileInfo getInfo() {
+    return mInfo;
+  }
+
+  /**
+   * @return the percentage of blocks in Tachyon memory tier storage, mutable
+   */
+  public int getInMemoryPercentage() {
+    return mInfo.getInMemoryPercentage();
+  }
+
+  /**
+   * @return the epoch time the entity referenced by this uri was last modified, mutable
+   */
+  public long getLastModificationTimeMs() {
+    return mInfo.getLastModificationTimeMs();
+  }
+
+  /**
+   * @return the length in bytes of the file, 0 for directories, mutable
+   */
+  public long getLength() {
+    return mInfo.getLength();
+  }
+
+  /**
+   * For example for the uri: tachyon://host:1000/foo/bar/baz, baz is the name
+   *
+   * @return the last path component of the entity referenced by this uri, mutable
+   */
+  public String getName() {
+    return mInfo.getName();
+  }
+
+  /**
+   * For example, for the uri: tachyon://host:1000/foo/bar/baz, the path is /foo/bar/baz
+   *
+   * @return the entire path component of the entity referenced by this uri, mutable
+   */
+  public String getPath() {
+    return mInfo.getPath();
+  }
+
+  /**
+   * @return the int representation of the ACL permissions of the entity referenced by this uri,
+   *         mutable
+   */
+  public int getPermission() {
+    return mInfo.getPermission();
+  }
+
+  /**
+   * @return the string representation of the persistence status, mutable
+   */
+  // TODO(calvin): Consider returning the enum if it is moved to common
+  public String getPersistenceState() {
+    return mInfo.getPersistenceState();
+  }
+
+  /**
+   * @return the time-to-live in milliseconds since the creation time of the entity referenced by
+   *         this uri, mutable
+   */
+  public long getTtl() {
+    return mInfo.getTtl();
+  }
+
+  /**
+   * @return the uri of the under storage location of the entity referenced by this uri, mutable
+   */
+  public String getUfsPath() {
```

Comment:
Will this method give the entire UFS URI, or just the path part? It seems like we distinguish URI vs path in Tachyon space, so do we do that as well in UFS space?